OpenKeychain/src/main/java/org/sufficientlysecure/keychain/keyimport/processing/ImportKeysListCloudLoader.java

```diff
@@ -83,8 +83,10 @@ public ImportKeysListCloudLoader(Context context, CloudLoaderState loaderState,
 
         // Now we have all the data needed to build the parcelable key ring for this key
         for (ImportKeysListEntry e : mEntryList) {
-            e.setParcelableKeyRing(ParcelableKeyRing.createFromReference(e.getFingerprint(), e.getKeyIdHex(),
-                    e.getKeybaseName(), e.getFbUsername()));
+            if (e.getParcelableKeyRing() == null) {
```

Comment:
Is this a good idea to introduce this here? I'm returning a key ring from search (search already includes entire key) and don't want to have it overridden here... If there is a better idea please tell.

Reply:
This stuff was never particularly pretty anyways. So yeah it's fine